lametro/views.py

```diff
@@ -717,6 +719,11 @@ def search(self):
             # Don't auto-escape my query! https://django-haystack.readthedocs.io/en/v2.4.1/searchqueryset_api.html#SearchQuerySet.filter
             sqs = sqs.filter_or(attachment_text=Raw(self.cleaned_data['q']))
 
+        if self.result_type == 'keyword':
+            sqs = sqs.exclude(topics__iexact=Exact(self.cleaned_data['q']))
```

Comment:
the `Exact` input type looks for the full phrase: https://django-haystack.readthedocs.io/en/v2.4.1/inputtypes.html#exact